pipeline/pipelines/core_generation_pipeline.py

```diff
@@ -55,13 +55,10 @@ class _GoCoreTaskFactory(code_gen.TaskFactoryBase):
     """
 
     def get_tasks(self, **kwargs):
-        return [
-            protoc_tasks.ProtoCodeGenTask('CoreProtoGen',
-                                          inject=kwargs),
-            protoc_tasks.GoExtractImportBaseTask('ExtractGoPackageName',
-                                                 inject=kwargs),
-            protoc_tasks.GoLangUpdateImportsTask('UpdateImports',
-                                                 inject=kwargs)]
+        tasks = [protoc_tasks.ProtoCodeGenTask,
+                 protoc_tasks.GoExtractImportBaseTask,
+                 protoc_tasks.GoLangUpdateImportsTask]
+        return task_utils.instantiate_tasks(tasks, kwargs)
```

Comment:
We have a confusing situation where instantiate_tasks can be called from both base pipeline code and task factory code. I think it would be simpler if we dealt only with non-instantiated tasks for as long as possible, and confined task instantiation to one place (or as few places as possible).